UnityProject/Assets/Scripts/Weapons/FireLemon.cs

```diff
@@ -0,0 +1,221 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Light2D;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using Mirror;
+
+/// <summary>
+/// Generic grenade base.
+/// </summary>
+[RequireComponent(typeof(Pickupable))]
+public class FireLemon : NetworkBehaviour, IPredictedInteractable<HandActivate>, IServerDespawn
+{
+	[Tooltip("Lowest potency explosion")]
+	public Explosion explosionPrefab1;
```

Comment:
Also, I don't really like these "explosion1", "explosion2" and etc, looks very messy.
I see two solutions for this:
1. Easy and not very preferable variant would be - turn this into array of explosions and calculate which index you want to take from it. (something like explosionIndex = explosionPrefabs.count * (plantPotency / potencyForMaxExplosion))
2. Second, and more preferable solution would be - to add SetUp method in Explosion.cs, where you could've set needed parameters for your explosion - radius, damage and etc. This way, you'd have 1 prefab and just array of setting for setting it up.

Reply:
How would I do the second solution? I'm very inexperienced with c#.

Reply:
add something like this to the explosions.cs 

```public void SetExplosionData(int damage = 150, float radius = 4f, bool unstableRadius = false, EffectShapeType explosionType = EffectShapeType.Circle, float shakeDistance = 8, int minDamage = 2, float maxEffectDuration = .25f, float minEffectDuration = .05f)
	{
		this.damage = damage;
		this.radius = radius;
		this.unstableRadius = unstableRadius;
		this.explosionType = explosionType;
		this.shakeDistance = shakeDistance;
		this.minDamage = minDamage;
		this.maxEffectDuration = maxEffectDuration;
		this.minEffectDuration = minEffectDuration;
	}```

Reply:
Then you call it and send the correct parameters when you spawn the prefab.

```
var explosionGO = Instantiate(explosionPrefab2, explosionMatrix.transform);
explosionGO.transform.position = worldPos;
var damage = 100;
var radius = 10f;
explosionGO.SetExplosionData([Send the stuff in here that you want to change eg damage, radius]);
explosionGO.Explode(explosionMatrix);
```

Reply:
If you need help ping me on discord.